modules/gapi/include/opencv2/gapi/rmat.hpp

```diff
@@ -70,23 +69,24 @@ class GAPI_EXPORTS RMat
         const std::vector<int>& dims() const { return m_desc.dims; }
         int cols() const { return m_desc.size.width; }
         int rows() const { return m_desc.size.height; }
-        int type() const { return CV_MAKE_TYPE(depth(), chan()); }
+        int type() const;
         int depth() const { return m_desc.depth; }
         int chan() const { return m_desc.chan; }
         size_t elemSize() const { return CV_ELEM_SIZE(type()); }
 
         template<typename T = uchar> T* ptr(int y = 0, int x = 0) {
-            return reinterpret_cast<T*>(m_data + m_step*y + x*CV_ELEM_SIZE(type()));
+            return reinterpret_cast<T*>(m_data + step()*y + x*CV_ELEM_SIZE(type()));
```

Comment:
What is `step()` here in the 2d case? what is it in, hm, 3d case? would this function work in 3d or 4d case?

Reply:
`step()` returns step value which should be applied when iterating on the first (with biggest order) dimension of the View. So it's a "width" step in 2d case as it was before, semantics unchanged, it's a "height" step in 3d case and "plane" step in 4d case.

Reply:
Regarding applicability in non-2d cases it's a good question if it even should be called in these cases with x != 0 and x != 0, maybe I should place an assert on that